lldb/source/Plugins/Process/gdb-remote/GDBRemoteRegisterContext.cpp

```diff
@@ -783,6 +783,11 @@ void GDBRemoteRegisterContext::AArch64Reconfigure() {
   std::optional<uint64_t> svg_reg_value;
   const RegisterInfo *svg_reg_info = m_reg_info_sp->GetRegisterInfo("svg");
   if (svg_reg_info) {
+    // When vg is written it is automatically made invalid. Writing vg will also
+    // change svg if we're in streaming mode but it will not be made invalid
+    // so do this manually so the following read gets the latest svg value.
```

Comment:
I am wondering whether we should find a way to make svg and vg inter dependent to make sure they are invalidated together whenever an update happens. Do you see a way within RegisterInfo class to link registers together. I dont fully remember but i believe this could be done.

Reply:
You're right there's `uint32_t *invalidate_regs;` in the register info, I didn't think to look at it.

The problem with that is it needs to know up front what the register number is. So we could make writes to svg invalidate vg since we know what vg's register number will be, but this is not needed since svg is read only. 

Making vg invalidate svg is harder because we don't know what number it will be, and it might not even exist.

Though, I could patch it in once we know we're adding the SME register set. Since after that we don't re-order registers, only change the size of some.

I'll give that a go. If patching it in works, it would lead to some redundant reads if you write vg during non-streaming mode, but it would be no worse than what I've got here I think.

Reply:
From what I've seen there's 2 scenarios to handle:
* Explicit write by the user to vg.
* We have just stopped and need to get the potentially new svg and vg.

The code as it was handled both but is really using mechanisms meant to handle the second scenario.

I have done what you suggest in a fixup commit, making vg invalidate svg. This will handle the first scenario specifically. If you approve of that I'll update the commit message prior to landing to explain that we have these two handlers for the 2 scenarios, and when/if they overlap, it is not a problem.

All of this is pessimistic, we assume any change to vg in any mode will invalidate svg. I think this is fine though because reading svg again is cheap, and getting the vector lengths mixed would waste a lot of user's time.

Reply:
Sorry for the delay on this PR. I agree with your approach.